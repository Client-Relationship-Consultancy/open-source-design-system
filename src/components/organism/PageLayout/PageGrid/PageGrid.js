import React from "react"
import styled from "styled-components"
import PropTypes from "prop-types"

const StyledGrid = styled.div`
  width: 100%;
  display: grid;
  grid-gap: 1rem;
  grid-template-columns: repeat(12, 1fr);
  grid-template-rows: ${props => `repeat(${props.rows}, 1fr)`};
`
StyledGrid.displayName = "StyledGrid"

const PageGrid = props => {
  const { children, rows, className, id } = props
  return (
    <StyledGrid id={id} rows={rows} className={className}>
      {children}
    </StyledGrid>
  )
}

PageGrid.defaultProps = {
  rows: 1,
}

PageGrid.propTypes = {
  children: PropTypes.node,
  rows: PropTypes.number,
  className: PropTypes.string,
  id: PropTypes.string,
}

export default PageGrid
