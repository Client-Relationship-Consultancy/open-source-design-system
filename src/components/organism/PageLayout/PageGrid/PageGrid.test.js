import React from "react"
import { shallow, mount } from "enzyme"
import "jest-styled-components"
import toJson from "enzyme-to-json"
import PageGrid from "./PageGrid"

describe("Page Grid Testing", () => {
  it("should match snapshot", () => {
    const component = shallow(<PageGrid>Hello World</PageGrid>)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  it("should render correct number of rows", () => {
    const component = mount(<PageGrid rows={3} />)
    expect(component).toHaveStyleRule("grid-template-rows", "repeat(3,1fr)")
  })
})
