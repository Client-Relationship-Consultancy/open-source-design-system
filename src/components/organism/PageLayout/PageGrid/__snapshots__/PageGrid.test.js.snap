// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Grid Testing should match snapshot 1`] = `
<StyledGrid
  rows={1}
>
  Hello World
</StyledGrid>
`;
